packages/ra-ui-materialui/src/input/ArrayInput/ArrayInput.tsx

```diff
@@ -141,6 +147,21 @@ export const ArrayInput = (props: ArrayInputProps) => {
 
     const { error } = getFieldState(finalSource, formState);
 
+    const sourceContext = React.useMemo<SourceContextValue>(
+        () => ({
+            getSource: (source: string) => {
+                return source ? `${finalSource}.${source}` : finalSource;
+            },
+            getLabel: (source: string) =>
+                parentSourceContext
+                    ? parentSourceContext.getLabel(
+                          `resources.${resourceFromProps}.fields.${source}`
```

Comment:
This is mysterious to me. Would you mind adding a comment to give an example expected label? 

Because to me, inside another ArrayInput, this will lead to labels like `resources.A.fields.B.0.resources.A.fields.C` instead of `resources.A.fields.B.0.fields.C`